src/validation.cpp

```diff
@@ -2653,6 +2660,12 @@ bool CChainState::ActivateBestChain(CValidationState &state, const CChainParams&
     // sanely for performance or correctness!
     AssertLockNotHeld(cs_main);
 
+    // ABC maintains a fair degree of expensive to calculate internal state
+    // because this function periodically releases cs_main so that it does not lock up other threads for too long
+    // during large re-orgs - and to allow for e.g. the callback queue to drain
```

Comment:
nit: comment is slightly wrong now, we dont release cs_main during large reorgs, only large connects.